src/Core/Framework/MessageQueue/Api/ConsumeMessagesController.php

```diff
@@ -134,10 +138,51 @@ public function consumeMessages(Request $request): JsonResponse
         $workerDispatcher->addSubscriber($this->dispatchPcntlSignalListener);
         $workerDispatcher->addSubscriber($this->earlyReturnListener);
 
+        if ($this->memoryLimit !== '-1') {
+            $workerDispatcher->addSubscriber(new StopWorkerOnMemoryLimitListener(
+                $this->convertToBytes($this->memoryLimit)
+            ));
+        }
+
         $worker = new Worker([$this->defaultTransportName => $receiver], $this->bus, $workerDispatcher);
 
         $worker->run(['sleep' => 50]);
 
         return $this->json(['handledMessages' => $listener->getHandledMessages()]);
     }
+
+    /**
+     * This code is the same as the memory limit implementation of Symfony's consume message command. Since we want to
+     * replicate the behaviour here we will be using the same implementation.
+     *
+     * See also:
+     * https://github.com/symfony/messenger/blob/5.4/Command/ConsumeMessagesCommand.php#L243-L266
```

Comment:
Can you link here a specific commit. The line markers will move

Reply:
I updated the comment 👍